wagtail/admin/templates/wagtailadmin/pages/listing/_list.html

```diff
@@ -24,7 +25,7 @@
                 {% page_permissions page as page_perms %}
                 <tr {% if ordering == "ord" %}id="page_{{ page.id|unlocalize }}" data-page-title="{{ page.get_admin_display_title }}"{% endif %} class="{% if not page.live %}unpublished{% endif %} {% block page_row_classname %}{% endblock %}">
                     {% if show_ordering_column %}
-                        <td class="ord">{% if orderable and ordering == "ord" %}<div class="handle icon icon-grip text-replace">{% trans 'Drag' %}</div>{% endif %}</td>
+                        <td class="ord">{% if orderable and ordering == "ord" %}<div class="handle icon icon-grip text-replace" tabindex="0" data-order-handle>{% trans 'Drag' %}</div>{% endif %}</td>
```

Comment:
Probably more complex - may not be worth the effort for this first round of a solution.

```suggestion
                        <td class="ord">
                            {% if orderable and ordering == "ord" %}
                                <div class="handle icon icon-grip text-replace" tabindex="0" aria-live="polite" data-order-handle>
                                    {% trans 'Drag' %}
                                    <span data-order-label>Item {{ forloop.counter }} of {{ pages|length }}</span>
                                </div>
                            {% endif %}
                        </td>
```